HighAvailabilityModule.Client.SQL/SQLMembershipClient.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace HighAvailabilityModule.Client.SQL
+{
+    using System;
+    using System.Threading.Tasks;
+    using System.Data;
+    using System.Data.SqlClient;
+
+    using HighAvailabilityModule.Interface;
+
+    public class SQLMembershipClient: IMembershipClient
+    {
+        public string Uuid { get; }
+
+        public string Utype { get; }
+
+        public string Uname { get; }
+
+        public TimeSpan OperationTimeout { get; set; }
+
+        public string ConStr { get; set; }
+
+        private string timeFormat = "yyyy-MM-dd HH:mm:ss.fff";
+
+        public SQLMembershipClient(string utype, string uname, TimeSpan operationTimeout)
+        {
+            this.Uuid = Guid.NewGuid().ToString();
+            this.Utype = utype;
+            this.Uname = uname;
+            this.OperationTimeout = operationTimeout;
+            this.ConStr = "server=.;database=HighAvailabilityWitness;Trusted_Connection=SSPI;Connect Timeout=" + Convert.ToInt32(Math.Ceiling(this.OperationTimeout.TotalSeconds)).ToString();
+        }
+
+        public async Task HeartBeatAsync(HeartBeatEntryDTO entryDTO)
+        {
+            SqlConnection con = new SqlConnection(this.ConStr);
+            string StoredProcedure = "dbo.HeartBeat";
+            SqlCommand comStr = new SqlCommand(StoredProcedure, con);
+            comStr.CommandType = CommandType.StoredProcedure;
+            comStr.CommandTimeout = Convert.ToInt32(Math.Ceiling(this.OperationTimeout.TotalSeconds));
+
+            comStr.Parameters.Add("@uuid", SqlDbType.NVarChar).Value = entryDTO.Uuid;
+            comStr.Parameters.Add("@utype", SqlDbType.NVarChar).Value = entryDTO.Utype;
+            comStr.Parameters.Add("@uname", SqlDbType.NVarChar).Value = entryDTO.Uname;
+            comStr.Parameters.Add("@lastSeenUuid", SqlDbType.NVarChar).Value = entryDTO.LastSeenEntry.Uuid;
+            comStr.Parameters.Add("@lastSeenUtype", SqlDbType.NVarChar).Value = entryDTO.LastSeenEntry.Utype;
+            comStr.Parameters.Add("@lastSeenTimeStamp", SqlDbType.DateTime).Value = entryDTO.LastSeenEntry.TimeStamp.ToString(this.timeFormat);
+
+            try
+            {
+                await con.OpenAsync();
+                await comStr.ExecuteNonQueryAsync();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"[{this.Uuid}] Error occured when sending heartbeat entry: {ex.ToString()}");
+                throw;
+            }
+            finally
+            {
+                con.Close();
+                con.Dispose();
+                comStr.Dispose();
+            }
+        }
+
+        public async Task<HeartBeatEntry> GetHeartBeatEntryAsync(string utype)
+        {
+            HeartBeatEntry heartBeatEntry;
+            SqlConnection con = new SqlConnection(this.ConStr);
+            string StoredProcedure = "dbo.GetHeartBeat";
```

Comment:
Make the stored procedure names as const strings and move under the class or some common file if they will be used in other places.